packages/playground/remote/service-worker.ts

```diff
@@ -26,9 +27,66 @@ if (!(self as any).document) {
 }
 
 reportServiceWorkerMetrics(self);
+const cache = new OfflineModeCache(buildVersion, self.location.hostname);
+
+/**
+ * For offline mode to work we need to cache all required assets.
+ *
+ * These assets are listed in the `/assets-required-for-offline-mode.json` file
+ * and contain JavaScript, CSS, and other assets required to load the site without
+ * making any network requests.
+ */
+cache.cacheOfflineModeAssets();
+
+/**
+ * Cleanup old cache.
+ *
+ * We cache data based on `buildVersion` which is updated whenever Playground is built.
+ * So when a new version of Playground is deployed, the service worker will remove the old cache and cache the new assets.
+ *
+ * If your build version doesn't change while developing locally check `buildVersionPlugin` for more details on how it's generated.
+ */
+cache.cleanup();
```

Comment:
The `.cleanup()` name make it seem like we're evicting the assets we just requested, how about calling it `.removeOutdatedFiles()` and calling if before `.cacheOfflineModeAssets()`?

Reply:
Also, let's make it clear we've skipped the `await` on purpose